pkg/analytics/analytics.go

```diff
@@ -24,59 +24,135 @@ SOFTWARE.
 package analytics
 
 import (
+	"bytes"
 	"crypto/md5"
 	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
 	"os"
-	"time"
+	"strings"
 
 	"github.com/denisbrodbeck/machineid"
-	"github.com/segmentio/analytics-go"
+	kuskBuild "github.com/kubeshop/kusk-gateway/pkg/build"
+)
+
+const (
+	gaUrl           = "https://www.google-analytics.com/mp/collect?measurement_id=%s&api_secret=%s"
+	gaValidationUrl = "https://www.google-analytics.com/debug/mp/collect?measurement_id=%s&api_secret=%s"
 )
 
 var (
-	TelemetryToken = "" // value needs to be passed with LDFLAG set to github.com/kubeshop/kusk-gateway/pkg/analytics.TelemetryToken
+	KuskGAMeasurementID = "" // value needs to be passed with LDFLAG set to github.com/kubeshop/kusk-gateway/pkg/analytics.KuskGAMeasurementID
+	KuskGAApiSecret     = "" // value needs to be passed with LDFLAG set to github.com/kubeshop/kusk-gateway/pkg/analytics.KuskGAApiSecret
 )
 
 func SendAnonymousInfo(event string) error {
-	track := analytics.Track{
-		AnonymousId: MachineID(),
-		Event:       "kusk-heartbeat",
-		Properties:  analytics.NewProperties().Set("event", event),
-		Timestamp:   time.Now(),
+	payload := Payload{
+		ClientID: MachineID(),
+		Events: []Event{
+			{
+				Name: event,
+				Params: Params{
+					EventCount:       1,
+					EventCategory:    "api-request",
+					AppName:          "kusk-gateway",
+					CustomDimensions: event,
+					DataSource:       "gateway",
+					AppVersion:       kuskBuild.Version,
+				},
+			},
+		},
 	}
-
-	return sendDataToGA(track)
+	return sendDataToGA(payload)
 }
 
 // SendAnonymouscmdInfo will send CLI event to GA
-func SendAnonymousCMDInfo() {
-	var event string
+func SendAnonymousCMDInfo() error {
+	event := "command"
 	command := []string{}
 	if len(os.Args) > 1 {
 		command = os.Args[1:]
 		event = command[0]
 	}
 
-	track := analytics.Track{
-		AnonymousId: MachineID(),
-		UserId:      MachineID(),
-		Event:       "kusk-cli",
-		Properties:  analytics.NewProperties().Set("event", event),
+	payload := Payload{
+		ClientID: MachineID(),
+		Events: []Event{
+			{
+				Name: event,
+				Params: Params{
+					EventCount:       1,
+					EventCategory:    "beacon",
+					AppName:          "kusk-cli",
+					CustomDimensions: strings.Join(command, " "),
+					AppVersion:       kuskBuild.Version,
+				},
+			}},
 	}
-	sendDataToGA(track)
-
+	return sendDataToGA(payload)
 }
 
-func sendDataToGA(track analytics.Track) error {
+func sendDataToGA(data Payload) error {
 	// if environment variable is set return and collect nothing
 	if _, ok := os.LookupEnv("KUSK_ANALYTICS_DISABLED"); ok {
 		return nil
 	}
+	sendValidationRequest(data)
```

Comment:
Are you intentionally not checking the `error` returned here?

Reply:
Yeah, if you don't need the output, perhaps change the function signature or intentionally ignore with `_, _ := sendValidationRequest(data)`

Reply:
It doesn't matter if it fails because we are not going to handle errors at all
Maybe we could add something when logging is set to debug?